File: app/src/main/java/ru/vincetti/vimoney/data/sqlite/ConfigDao.java
package ru.vincetti.vimoney.data.sqlite;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ru.vincetti.vimoney.data.models.ConfigModel;

@Dao
public interface ConfigDao {
    @Query("SELECT * FROM config")
    LiveData<List<ConfigModel>> loadConfigs();

    @Query("SELECT * FROM config WHERE key_name = :key")
    LiveData<ConfigModel> loadConfigByKey(String key);

    @Insert
    void insertConfig(ConfigModel conf);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateConfig(ConfigModel conf);
}

File: app/src/main/java/ru/vincetti/vimoney/data/sqlite/ConfigDao.kt
package ru.vincetti.vimoney.data.sqlite

import androidx.room.*
import ru.vincetti.vimoney.data.models.ConfigModel

@Dao
interface ConfigDao {

    @Query("SELECT * FROM config WHERE key_name = :key LIMIT 1")
    suspend fun loadConfigByKey(key: String): ConfigModel?

    @Insert
    suspend fun insertConfig(conf: ConfigModel)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    suspend fun updateConfig(conf: ConfigModel)

}
